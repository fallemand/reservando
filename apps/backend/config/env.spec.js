describe("environmental variables", () => {
  const ENV = {
    BUILD: "false",
    NODE_ENV: "production",
    FIREBASE_PROJECT_ID: "__PROJECT_ID__",
    FIREBASE_DATABASE_URL: "__DATABASE_URL__",
    GOOGLE_APPLICATION_CREDENTIALS: '{"__KEY__": "__VALUE__"}',
  };

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...ENV };
  });

  test("should generate map of values", () => {
    // eslint-disable-next-line global-require
    const env = require("./env");
    expect(env).toMatchSnapshot();
  });
});
