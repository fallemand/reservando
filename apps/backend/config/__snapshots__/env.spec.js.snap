// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`environmental variables should generate map of values 1`] = `
Object {
  "default": Object {
    "env": "production",
    "firebaseConfig": Object {
      "databaseURL": "__DATABASE_URL__",
      "googleAppCredentials": Object {
        "__KEY__": "__VALUE__",
      },
      "projectId": "__PROJECT_ID__",
    },
    "isDevelopment": false,
    "isProduction": true,
    "isTesting": false,
    "port": 3005,
  },
}
`;
